flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,18 +185,45 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
+
+		while(true) {
```

Comment:
It seems that this loop is now executed also for the YARN session. In the past, this was used only for "per job" yarn clusters, because there we needed to have all slots available to submit the job.

Now, this loop is "blocking" the YARN session CLI if not all TaskManager's are connecting. Its not a big deal, but a change in the semantics.